### ISSUE TITLE
Conv1d - Conv2d register_backward_hook inconsistency

### ISSUE THREAD

**Issue by vmoens** [pytorch/pytorch#23592]
## 🐛 Bug

The method `register_backward_hook` on `Conv1d` and `Conv2d` does not behave the same way.

## To Reproduce
A simple test is to run
```
def fun(module,grad_in,grad_out):
    print('grad_in')
    print([_grad_in.shape for _grad_in in grad_in if _grad_in is not None])     # add break point here
    print('grad_out')
    print([_grad_out.shape for _grad_out in grad_out if _grad_out is not None])     # add break point here

net = nn.Sequential(nn.Conv2d(1,1,3,), nn.ReLU(), nn.Conv2d(1,1,3,))

net[0].register_backward_hook(fun)
net[2].register_backward_hook(fun)

x = torch.randn(1,1,15,15,requires_grad=True)
l = net(x)

l.backward(torch.ones_like(l))
```
and then
```
net = nn.Sequential(nn.Conv1d(1,1,3,), nn.ReLU(), nn.Conv1d(1,1,3,))

net[0].register_backward_hook(fun)
net[2].register_backward_hook(fun)

x = torch.randn(1,1,15,requires_grad=True)
l = net(x)

l.backward(torch.ones_like(l))
```
for Conv1d.

The output of the first is 
```
grad_in
[torch.Size([1, 1, 13, 13]), torch.Size([1, 1, 3, 3]), torch.Size([1])]
grad_out
[torch.Size([1, 1, 11, 11])]
grad_in
[torch.Size([1, 1, 15, 15]), torch.Size([1, 1, 3, 3]), torch.Size([1])]
grad_out
[torch.Size([1, 1, 13, 13])]
```
which is ok, but the second (`Conv1d`) displays something completely different and hard to interpret:
```
grad_in
[torch.Size([1, 1, 1, 11])]
grad_out
[torch.Size([1, 1, 11])]
grad_in
[torch.Size([1, 1, 1, 13])]
grad_out
[torch.Size([1, 1, 13])]
```
Of course a quick fix can be to use `Conv2d(n_input, n_output, (kernel, 1))` and unsqueeze the input tensor to 4d, but that's a bit dirty.

## Expected behavior

I guess `Conv1d`'s `grad_in` should be a tuple with gradients of input, weights and biases.
 
## Environment

Collecting environment information...
PyTorch version: 1.1.0.post2
Is debug build: No
CUDA used to build PyTorch: None

OS: Mac OSX 10.13.2
GCC version: Could not collect
CMake version: Could not collect

Python version: 3.5
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

**Comment by albanD** [comment]
Hi,

Unfortunately this is because the backward hooks on nn.Modules are not working as expected at the moment.
This is tracked in #598 
Feel free to re-open this issue when the one above will be fixed if you still see problems.
The recommanded workaround is to use `register_hook()` on each Tensor for which you want the gradients.
